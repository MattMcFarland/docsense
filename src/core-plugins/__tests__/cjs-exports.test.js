const testFactory = require('./utils/testFactory');
const cjsExportsPlugin = require('../cjs-exports');
const tests = [
  'module.exports = function () {}',
  'exports = function () {}',
  'module.exports = foo()',
  'exports = foo()',
  'module.exports = new Foo()',
  'exports = new Foo()',
  'module.exports = new a.b.c.Foo()',
  'exports = new a.b.c.Foo()',
  'module.exports = []',
  'exports = []',
  'module.exports = () => {}',
  'exports = () => {}',
  'module.exports = fn.call()',
  'exports = fn.call()',
  'module.exports = fn.apply()',
  'exports = fn.apply()',
  '(function () { module.exports = foo })()',
  'module.exports.name = function () {}',
  'exports.name = function () {}',
  'module.exports.name = foo()',
  'exports.name = foo()',
  'module.exports.name = []',
  'exports.name = []',
  'module.exports.name = () => {}',
  'exports.name = () => {}',
  'module.exports.name = fn.call()',
  'exports.name = fn.call()',
  'module.exports.name = fn.apply()',
  'exports.name = fn.apply()',
];
const withComments = [
  `/**
  * Create Super Hero
  * @param {object} param
  * @param {string} param.name - the name of your super hero
  * @param {string} param.ability - your hero's special ability
  * @returns {SuperHero} your new super hero!!
  */
 module.exports.createSuperHero = ({name, ability}) => {}`,
  `/**
 * Create Super Hero
 * @param {object} param
 * @param {string} param.name - the name of your super hero
 * @param {string} param.ability - your hero's special ability
 * @returns {SuperHero} your new super hero!!
 */
module.exports = ({name, ability}) => {}`,
  `/**
* Create Super Hero
* @param {object} param
* @param {string} param.name - the name of your super hero
* @param {string} param.ability - your hero's special ability
* @returns {SuperHero} your new super hero!!
*/
exports = ({name, ability}) => {}`,
  `/**
* Create Super Hero
* @param {object} param
* @param {string} param.name - the name of your super hero
* @param {string} param.ability - your hero's special ability
* @returns {SuperHero} your new super hero!!
*/
module.exports = function({name, ability}) {}`,
];
const emptyTests = [
  'var exports = 4',
  'let foo = { exports: 4}',
  'const foo = bar.exports',
  'const foo = derp.exports',
  'const foo = bar.exports.baz',
  'foo.map(exports => exports.module)',
  'foo.map(module => module.exports = bar)',
  'module.other.exports = bar',
  'module.other.exports = { exports }',
  'module.other.exports.foo = bar',
  'module.other.exports.foo = { exports }',
  'some.module.exports = baz',
  'some.module.exports = { exports }',
  'some.exports = baz',
  'some.module.exports.foo = baz',
  'some.module.exports.foo = { exports }',
  'some.exports.foo = baz',
  'some.exports.foo = { exports }',
  'some.module.other.exports = bar',
  'some.module.other.exports = { exports }',
  'some.other.exports = bar',
  'some.other.exports = { exports }',
  'some.module.other.exports.foo = baz',
  'some.module.other.exports.foo = { exports }',
  'some.other.exports.foo = baz',
  'some.other.exports.foo = { exports }',
];
const suites = {
  OK: tests,
  COMMENTS: withComments,
  EMPTY: emptyTests,
};
describe('Core: common.js', () =>
  testFactory({
    plugin: cjsExportsPlugin,
    suites,
  }));
