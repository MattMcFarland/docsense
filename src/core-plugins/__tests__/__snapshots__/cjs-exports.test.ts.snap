// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core: common.js COMMENTS /**
  * Create Super Hero
  * @param {object} param
  * @param {string} param.name - the name of your super hero
  * @param {string} param.ability - your hero's special ability
  * @returns {SuperHero} your new super hero!!
  */
 module.exports.createSuperHero = ({name, ability}) => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "createSuperHero",
    "file_id": "__TEST__",
    "function_id": "anonymous@8:34",
    "jsdoc": Array [
      Object {
        "description": "Create Super Hero",
        "tags": Array [
          Object {
            "description": null,
            "name": "param",
            "title": "param",
            "type": Object {
              "name": "object",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "the name of your super hero",
            "name": "param.name",
            "title": "param",
            "type": Object {
              "name": "string",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "your hero's special ability",
            "name": "param.ability",
            "title": "param",
            "type": Object {
              "name": "string",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "your new super hero!!",
            "title": "returns",
            "type": Object {
              "name": "SuperHero",
              "type": "NameExpression",
            },
          },
        ],
      },
    ],
  },
]
`;

exports[`Core: common.js COMMENTS /**
 * Create Super Hero
 * @param {object} param
 * @param {string} param.name - the name of your super hero
 * @param {string} param.ability - your hero's special ability
 * @returns {SuperHero} your new super hero!!
 */
module.exports = ({name, ability}) => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "function_id": "anonymous@8:17",
    "jsdoc": Array [
      Object {
        "description": "Create Super Hero",
        "tags": Array [
          Object {
            "description": null,
            "name": "param",
            "title": "param",
            "type": Object {
              "name": "object",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "the name of your super hero",
            "name": "param.name",
            "title": "param",
            "type": Object {
              "name": "string",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "your hero's special ability",
            "name": "param.ability",
            "title": "param",
            "type": Object {
              "name": "string",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "your new super hero!!",
            "title": "returns",
            "type": Object {
              "name": "SuperHero",
              "type": "NameExpression",
            },
          },
        ],
      },
    ],
  },
]
`;

exports[`Core: common.js COMMENTS /**
* Create Super Hero
* @param {object} param
* @param {string} param.name - the name of your super hero
* @param {string} param.ability - your hero's special ability
* @returns {SuperHero} your new super hero!!
*/
exports = ({name, ability}) => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "function_id": "anonymous@8:10",
    "jsdoc": Array [
      Object {
        "description": "Create Super Hero",
        "tags": Array [
          Object {
            "description": null,
            "name": "param",
            "title": "param",
            "type": Object {
              "name": "object",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "the name of your super hero",
            "name": "param.name",
            "title": "param",
            "type": Object {
              "name": "string",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "your hero's special ability",
            "name": "param.ability",
            "title": "param",
            "type": Object {
              "name": "string",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "your new super hero!!",
            "title": "returns",
            "type": Object {
              "name": "SuperHero",
              "type": "NameExpression",
            },
          },
        ],
      },
    ],
  },
]
`;

exports[`Core: common.js COMMENTS /**
* Create Super Hero
* @param {object} param
* @param {string} param.name - the name of your super hero
* @param {string} param.ability - your hero's special ability
* @returns {SuperHero} your new super hero!!
*/
module.exports = function({name, ability}) {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "function_id": "anonymous@8:17",
    "jsdoc": Array [
      Object {
        "description": "Create Super Hero",
        "tags": Array [
          Object {
            "description": null,
            "name": "param",
            "title": "param",
            "type": Object {
              "name": "object",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "the name of your super hero",
            "name": "param.name",
            "title": "param",
            "type": Object {
              "name": "string",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "your hero's special ability",
            "name": "param.ability",
            "title": "param",
            "type": Object {
              "name": "string",
              "type": "NameExpression",
            },
          },
          Object {
            "description": "your new super hero!!",
            "title": "returns",
            "type": Object {
              "name": "SuperHero",
              "type": "NameExpression",
            },
          },
        ],
      },
    ],
  },
]
`;

exports[`Core: common.js EMPTY const foo = bar.exports 1`] = `Array []`;

exports[`Core: common.js EMPTY const foo = bar.exports.baz 1`] = `Array []`;

exports[`Core: common.js EMPTY const foo = derp.exports 1`] = `Array []`;

exports[`Core: common.js EMPTY foo.map(exports => exports.module) 1`] = `Array []`;

exports[`Core: common.js EMPTY foo.map(module => module.exports = bar) 1`] = `Array []`;

exports[`Core: common.js EMPTY let foo = { exports: 4} 1`] = `Array []`;

exports[`Core: common.js EMPTY module.other.exports = { exports } 1`] = `Array []`;

exports[`Core: common.js EMPTY module.other.exports = bar 1`] = `Array []`;

exports[`Core: common.js EMPTY module.other.exports.foo = { exports } 1`] = `Array []`;

exports[`Core: common.js EMPTY module.other.exports.foo = bar 1`] = `Array []`;

exports[`Core: common.js EMPTY some.exports = baz 1`] = `Array []`;

exports[`Core: common.js EMPTY some.exports.foo = { exports } 1`] = `Array []`;

exports[`Core: common.js EMPTY some.exports.foo = baz 1`] = `Array []`;

exports[`Core: common.js EMPTY some.module.exports = { exports } 1`] = `Array []`;

exports[`Core: common.js EMPTY some.module.exports = baz 1`] = `Array []`;

exports[`Core: common.js EMPTY some.module.exports.foo = { exports } 1`] = `Array []`;

exports[`Core: common.js EMPTY some.module.exports.foo = baz 1`] = `Array []`;

exports[`Core: common.js EMPTY some.module.other.exports = { exports } 1`] = `Array []`;

exports[`Core: common.js EMPTY some.module.other.exports = bar 1`] = `Array []`;

exports[`Core: common.js EMPTY some.module.other.exports.foo = { exports } 1`] = `Array []`;

exports[`Core: common.js EMPTY some.module.other.exports.foo = baz 1`] = `Array []`;

exports[`Core: common.js EMPTY some.other.exports = { exports } 1`] = `Array []`;

exports[`Core: common.js EMPTY some.other.exports = bar 1`] = `Array []`;

exports[`Core: common.js EMPTY some.other.exports.foo = { exports } 1`] = `Array []`;

exports[`Core: common.js EMPTY some.other.exports.foo = baz 1`] = `Array []`;

exports[`Core: common.js EMPTY var exports = 4 1`] = `Array []`;

exports[`Core: common.js OK (function () { module.exports = foo })() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports = () => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "function_id": "anonymous@1:10",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports = [] 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports = fn.apply() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports = fn.call() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports = foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports = function () {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "function_id": "anonymous@1:10",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports = new Foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports = new a.b.c.Foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports.name = () => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "function_id": "anonymous@1:15",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports.name = [] 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports.name = fn.apply() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports.name = fn.call() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports.name = foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK exports.name = function () {} 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "function_id": "anonymous@1:15",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports = () => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "function_id": "anonymous@1:17",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports = [] 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports = fn.apply() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports = fn.call() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports = foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports = function () {} 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "function_id": "anonymous@1:17",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports = new Foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports = new a.b.c.Foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "default",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports.name = () => {} 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "function_id": "anonymous@1:22",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports.name = [] 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports.name = fn.apply() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports.name = fn.call() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports.name = foo() 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "jsdoc": undefined,
  },
]
`;

exports[`Core: common.js OK module.exports.name = function () {} 1`] = `
Array [
  Object {
    "cjsExports_id": "name",
    "file_id": "__TEST__",
    "function_id": "anonymous@1:22",
    "jsdoc": undefined,
  },
]
`;
